resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -202,9 +203,13 @@ private[spark] class Client(
           }
         }
       }
-    } else {
-      logInfo(s"Deployed Spark application ${conf.appName} with application ID ${conf.appId} " +
-        s"and submission ID $sId into Kubernetes")
+      // no-test case exit with the driver's exit code
+      if (!sys.props.get(IS_TESTING.key).getOrElse("false").toBoolean) {
+        watcher.getDriverExitCode().foreach { exitCode =>
```

Comment:
This PR seems to propose to introduce a new `SparkException` point to this class. In general, it's not good, as you know.